diff --git a/src/Plugin/Field/FieldFormatter/AiAudioFieldFormatter.php b/src/Plugin/Field/FieldFormatter/AiAudioFieldFormatter.php
new file mode 100644
index 0000000000000000000000000000000000000000..c385e5cd80a2888a1ee7d221cb390be11305b69a
--- /dev/null
+++ b/src/Plugin/Field/FieldFormatter/AiAudioFieldFormatter.php
@@ -0,0 +1,19 @@
+<?php
+
+namespace Drupal\ai_audio_field\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\Attribute\FieldFormatter;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\file\Plugin\Field\FieldFormatter\GenericFileFormatter;
+
+/**
+ * Plugin implementation of the 'ai_audio_file' formatter.
+ */
+#[FieldFormatter(
+  id: 'ai_audio_file',
+  label: new TranslatableMarkup('AI Audio Generic file'),
+  field_types: [
+    'ai_audio_file',
+  ],
+)]
+class AiAudioFieldFormatter extends GenericFileFormatter {}
diff --git a/src/Plugin/Field/FieldType/AiAudioField.php b/src/Plugin/Field/FieldType/AiAudioField.php
index 4ae0ef313085803a1e265da0c075775c9d1b3d6a..86c0de7e5f98079166fbc62803ffa77dade00144 100644
--- a/src/Plugin/Field/FieldType/AiAudioField.php
+++ b/src/Plugin/Field/FieldType/AiAudioField.php
@@ -26,7 +26,7 @@ use Drupal\file\Plugin\Field\FieldType\FileItem;
   ],
   category: "file_upload",
   default_widget: "ai_audio_field_widget",
-  default_formatter: "file_default",
+  default_formatter: "ai_audio_file",
   list_class: AiAudioFieldItem::class,
   constraints: ["ReferenceAccess" => []],
   column_groups: [
